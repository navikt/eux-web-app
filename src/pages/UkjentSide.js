import PT from 'prop-types';
import React from 'react';
import Ui from 'eessi-pensjon-ui'
import './UkjentSide.css';

/* eslint arrow-body-style:off */
const UkjentSide = ({ location }) => {
  /*
  const logdata = {
    message: 'Ukjent Side',
    data: {
      url: location.pathname,
    },
  };
  window.frontendlogger.error(logdata);
  */
  return (
    <Ui.Nav.AlertStripe type="stopp" className="ukjentSide">
      <Ui.Nav.Systemtittel>Denne siden finnes ikke: &quot;{location.pathname}&quot;.</Ui.Nav.Systemtittel>
      <p>Dersom du ble sendt hit fra Gosys eller et annet Nav-system, ta kontakt med driftsansvarlig.</p>
      <Ui.Nav.Lenke href="/" ariaLabel="Navigasjonslink tilbake til forsiden">
        Klikk her for å gå tilbake til forsiden
      </Ui.Nav.Lenke>
    </Ui.Nav.AlertStripe>
  );
};

UkjentSide.propTypes = {
  location: PT.object.isRequired,
};

export default UkjentSide;
