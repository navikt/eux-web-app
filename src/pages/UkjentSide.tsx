import { Container, Content, Margin } from 'nav-hoykontrast'
import TopContainer from 'components/TopContainer/TopContainer'
import Alertstripe from 'nav-frontend-alertstriper'
import Lenke from 'nav-frontend-lenker'
import { Systemtittel } from 'nav-frontend-typografi'
import PT from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'

export interface UkjentSideProps {
  location: Location
}

const UkjentSide: React.FC<UkjentSideProps> = ({ location }: UkjentSideProps): JSX.Element => {
  const { t } = useTranslation()
  return (
    <TopContainer>
      <Container>
        <Margin />
        <Content>
          <Alertstripe type='advarsel'>
            <Systemtittel>
              {t('message:error-unknownPage-cantFindRoute', { pathname: location.pathname })}
            </Systemtittel>
          </Alertstripe>
          <p>{t('message:error-unknownPage-description')}</p>
          <Lenke href='/' ariaLabel={t('message:error-unknownPage-linkToRoot-ariaLabel')}>
            {t('message:error-unknownPage-linkToRoot')}
          </Lenke>
        </Content>
        <Margin />
      </Container>
    </TopContainer>
  )
}

UkjentSide.propTypes = {
  location: PT.any.isRequired
}

export default UkjentSide
