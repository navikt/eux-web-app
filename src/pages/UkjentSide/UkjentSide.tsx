import { Container, Content, Margin } from 'nav-hoykontrast'
import TopContainer from 'components/TopContainer/TopContainer'
import { Alert, Link, Heading } from '@navikt/ds-react'
import PT from 'prop-types'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { useLocation } from 'react-router-dom'

const UkjentSide: React.FC = (): JSX.Element => {
  const { t } = useTranslation()
  const location = useLocation()
  return (
    <TopContainer title={t('app:page-title-unknown')}>
      <Container>
        <Margin />
        <Content>
          <Alert variant='warning'>
            <Heading size='medium'>
              {t('message:error-unknownPage-cantFindRoute', { pathname: location.pathname })}
            </Heading>
          </Alert>
          <p>{t('message:error-unknownPage-description')}</p>
          <Link
            href='/'
            ariaLabel={t('message:error-unknownPage-linkToRoot-ariaLabel')}
          >
            {t('message:error-unknownPage-linkToRoot')}
          </Link>
        </Content>
        <Margin />
      </Container>
    </TopContainer>
  )
}

UkjentSide.propTypes = {
  location: PT.any.isRequired
}

export default UkjentSide
