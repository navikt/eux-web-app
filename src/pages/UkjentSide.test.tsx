import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import UkjentSide, { UkjentSideProps } from './UkjentSide'
jest.mock('components/TopContainer/TopContainer', () => ({children}: any) => (<div className='mock-topContainer'>{children}</div>))

describe('pages/OpprettSak/OpprettSak', () => {
  let wrapper: ReactWrapper
  const initialMockProps: UkjentSideProps = {
    location: {
      host: '',
      hash: '',
      hostname: '',
      href: '',
      origin: '',
      pathname: '',
      port: '',
      protocol: '',
      search: ''
    } as Location
  }

  beforeEach(() => {
    wrapper = mount(<UkjentSide {...initialMockProps}/>)
  })

  afterEach(() => {
    wrapper.unmount()
  })


  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('HTML structure', () => {
    expect(wrapper.exists('.p-ukjentSide')).toBeTruthy()
  })
})
