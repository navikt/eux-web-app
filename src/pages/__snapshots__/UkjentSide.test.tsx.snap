// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/OpprettSak/OpprettSak Renders 1`] = `
<UkjentSide
  location={
    Object {
      "hash": "",
      "host": "",
      "hostname": "",
      "href": "",
      "origin": "",
      "pathname": "",
      "port": "",
      "protocol": "",
      "search": "",
    }
  }
>
  <Component
    className="p-ukjentSide"
  >
    <div
      className="mock-topContainer"
    >
      <Row
        className="m-0"
      >
        <div
          className="row m-0"
        >
          <div
            className="col-sm-1"
          />
          <div
            className="col-sm-10 m-4"
          >
            <AlertStripe
              form="default"
              size="1.5em"
              type="advarsel"
            >
              <div
                className="alertstripe alertstripe--advarsel"
              >
                <span
                  className="alertstripe__ikon"
                >
                  <span
                    className="sr-only"
                  >
                    advarsel
                  </span>
                  <Ikon
                    kind="advarsel-sirkel-fyll"
                    size="1.5em"
                  >
                    <svg
                      focusable="false"
                      height="1.5em"
                      kind="advarsel-sirkel-fyll"
                      viewBox="0 0 24 24"
                      width="1.5em"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                      >
                        <path
                          d="M12.205-.004l-.214.002a12.225 12.225 0 0 0-8.517 3.659C1.179 5.977-.053 9.013.002 12.208c.115 6.613 5.296 11.793 11.795 11.793l.212-.002c6.726-.116 12.105-5.595 11.99-12.21C23.883 5.178 18.702-.003 12.204-.003z"
                          fill="#FFA733"
                          fillRule="nonzero"
                        />
                        <path
                          d="M12.027 19H12A1.499 1.499 0 0 1 11.973 16L12 16a1.501 1.501 0 0 1 .027 3z"
                          fill="#3E3832"
                        />
                        <path
                          d="M12 5a1 1 0 0 1 1 1v7a1 1 0 0 1-2 0V6a1 1 0 0 1 1-1z"
                          fill="#3E3832"
                          fillRule="nonzero"
                        />
                      </g>
                    </svg>
                  </Ikon>
                </span>
                <Normaltekst
                  className="alertstripe__tekst"
                  tag="div"
                >
                  <n
                    className="alertstripe__tekst"
                    tag="div"
                    type="normaltekst"
                  >
                    <div
                      className="typo-normal alertstripe__tekst"
                    >
                      <Systemtittel>
                        <n
                          type="systemtittel"
                        >
                          <h2
                            className="typo-systemtittel"
                          >
                            ui:error-unknownPage-cantFindRoute
                          </h2>
                        </n>
                      </Systemtittel>
                    </div>
                  </n>
                </Normaltekst>
              </div>
            </AlertStripe>
            <p>
              ui:error-unknownPage-description
            </p>
            <Lenke
              ariaLabel="ui:error-unknownPage-linkToRoot-ariaLabel"
              href="/"
            >
              <a
                aria-label="ui:error-unknownPage-linkToRoot-ariaLabel"
                className="lenke"
                href="/"
              >
                ui:error-unknownPage-linkToRoot
              </a>
            </Lenke>
          </div>
          <div
            className="col-sm-1"
          />
        </div>
      </Row>
    </div>
  </Component>
</UkjentSide>
`;
